plugins/inputs/unbound/unbound.go

```diff
@@ -106,6 +94,17 @@ func (s *Unbound) Gather(acc telegraf.Accumulator) error {
 					fieldsThreads[threadID][field] = fieldValue
 				}
 			}
+		} else if strings.HasPrefix(stat, "histogram") {
+			statTokens := strings.Split(stat, ".")
+			if s.Histogram && len(statTokens) > 1 {
+				lbound, err := strconv.ParseFloat(strings.Join(statTokens[1:3], "."), 64)
+				if err != nil {
+					acc.AddError(fmt.Errorf("expected a numeric value for the histogram bucket lower bound: %s", strings.Join(statTokens[1:3], ".")))
+					continue
+				}
+				field := fmt.Sprintf("%s_%f", statTokens[0], lbound)
+				fields[field] = fieldValue
+			}
 		} else {
```

Comment:
Here my suggestion with the simpler code and the leading zeros removed
```suggestion
		} else if suffix, found := strings.CutPrefix(x, "histogram."); found {
			if s.Histogram {			
				suffix = strings.TrimLeft(suffix, "0")
				fields["histogram_"+suffix] = fieldValue
			}
		} else {
```